### ISSUE TITLE
Support system-manager-based activation

### ISSUE THREAD

**Issue by grigorescu** [actor-framework/actor-framework#1221]
Some system managers (systemd, launchd) support socket-based activation, in the old inetd style. The system manager listens on a socket (AF_INET, AF_UNIX, or others), and then when something comes in, it can start the appropriate process and pass the fd to that process.

The systemd call for this is [sd_listen_fds](https://www.freedesktop.org/software/systemd/man/sd_listen_fds.html), for instance.

My goal is to better integrate my application with the system's service manager -- I have some endpoints that listen on particular ports, and let the service manager start and stop them as needed. This also would let me orchestrate a service that runs on multiple systems.

Does supporting this type of activation make sense?

**Comment by Neverlord** [comment]
> Does supporting this type of activation make sense?

What kind of support would you been looking for? What service are you providing on that socket?

- Are you implementing the socket communication on your own? Then I suppose CAF is for organizing background work - and then I don't think there's something we would need to add to CAF for this use case. Once your application boots up, you can create the actor system as usual.
- Do you want to have CAF take care of the socket returned by `sd_listen_fds,`, i.e., bind an actor to that socket so that others can connect via `remote_actor`? This is already possible, although there's no API in place to make this use case convenient yet. Would "supporting this type of activation" essentially boil down to a function that hands a socket over to CAF and binds an actor to it?

**Comment by grigorescu** [comment]
> * Do you want to have CAF take care of the socket returned by `sd_listen_fds,`, i.e., bind an actor to that socket so that others can connect via `remote_actor`? This is already possible, although there's no API in place to make this use case convenient yet. Would "supporting this type of activation" essentially boil down to a function that hands a socket over to CAF and binds an actor to it?

Yeah, exactly. I think that function to hand a socket into CAF and bind an actor to it is what would be needed.

My immediate use-case is a multi-system Zeek cluster, and having systemd handle cluster management rather than zeekctl.
